### PR TITLE
Feature: add support for serving from custom path

### PR DESCRIPTION
**Pull request by Squixx**
### Description
I want to run dependency track frontend on a path instead of `/`, but doing this currently would require to do a full rebuild of the frontend. This PR could be a possible solution direction. 

I've seen the reference to:

```
 // Relative paths cannot be supported. Research by @nscur0 - https://owasp.slack.com/archives/CTC03GX9S/p1608400149085400
  publicPath: '/',
``` 

However I can not open this link, so am oblivious to any previous research

I'll update the docs once it's clear this is the way forward

### Addressed Issue

https://github.com/DependencyTrack/dependency-track/discussions/1514

### Additional Details

setting the `BASE_HREF` env variable allows you to set a path 

### Checklist

- [x] I have read and understand the [contributing guidelines](https://github.com/DependencyTrack/dependency-track/blob/master/CONTRIBUTING.md#pull-requests)
- [ ] This PR introduces new or alters existing behavior, and I have updated the [documentation](https://github.com/DependencyTrack/dependency-track/tree/master/docs/_docs) accordingly

### CODE REVIEW

**Comment by fnxpt** [comment]
similar to #23

**Comment by Squixx** [comment]
> similar to #23

Does that work with oidc? This pr also has working oidc, but I'd be happy if either could move forward

**Comment by nscuro** [comment]
Sorry for the long wait @Squixx, I'll have a look at this next and have assigned it to v4.12.

File: docker/Dockerfile.alpine [hunk 1]
```diff
@@ -9,8 +9,8 @@ ENV TZ=Etc/UTC \
     LANG=C.UTF-8 \
     # Set default settings that may get overridden to empty values by
     # the entrypoint script, if not explicitly provided by the user
-    OIDC_SCOPE="openid profile email"
-
+    OIDC_SCOPE="openid profile email" \
+    BASE_HREF="/"
```
Comment (nscuro) [inline docker/Dockerfile.alpine:13]: Consider renaming this to `BASE_URL` or `BASE_PATH` so it's more in line with how the other options are worded (i.e. `API_BASE_URL`).

File: docker/Dockerfile.alpine [hunk 2]
```diff
@@ -29,10 +29,13 @@ RUN chown -R 101:0 ${APP_DIR} \
 # See https://github.com/nginxinc/docker-nginx-unprivileged/blob/main/stable/alpine/Dockerfile#L139
 USER 101
 
+RUN mkdir /etc/nginx/templates
```
Comment (nscuro) [inline docker/Dockerfile.alpine:32]: Ha, TIL about [config templates](https://devopsian.net/p/nginx-config-template-with-environment-vars/), nice!

**Review by nscuro — COMMENTED** [review]
Again sorry for the delay! Overall this looks good to me. 

Just a minor nitpick about naming of the env var.

Also, it would be good to add a few words about how to configure this in the docs: https://github.com/DependencyTrack/dependency-track/blob/master/docs/_docs/getting-started/configuration.md#frontend

File: docker/Dockerfile.alpine (hunk 1 above)
Comment (Squixx) [reply docker/Dockerfile.alpine]: BASE_PATH probably makes most sense, and would trigger anyone to put in the full URL. Updated

**Comment by Squixx** [comment]
> Again sorry for the delay! Overall this looks good to me.
> 
> Just a minor nitpick about naming of the env var.
> 
> Also, it would be good to add a few words about how to configure this in the docs: https://github.com/DependencyTrack/dependency-track/blob/master/docs/_docs/getting-started/configuration.md#frontend

added docs in https://github.com/DependencyTrack/dependency-track/pull/3899

**Review by nscuro — APPROVED** [review]

**Pull request merged**